libevm/VM.cpp

```diff
@@ -233,6 +233,24 @@ void VM::interpretCases()
 		}
 		BREAK
 
+		CASE(REVERT)
+		{
+			// Pre-metropolis
+			if (!m_schedule->haveRevert)
+				throwBadInstruction();
+
+			m_newMemSize = memNeed(*m_SP, *(m_SP - 1));
+			updateMem();
+			ON_OP();
+			updateIOGas();
+
+			uint64_t b = (uint64_t)*m_SP--;
+			uint64_t s = (uint64_t)*m_SP--;
+			owning_bytes_ref output{ move(m_mem), b, s };
```

Comment:
I think we should use `pair<bool, bytes>` (extended `std::optional`), as @chriseth did in precompiled already. We can name this time, and add convenient methods similar to `std::optional`. I think nothing from boost/std directly matches our case here.

If this is a lot refactoring, it can be done in a later PR. Ideally, implement the return type change first, then EIP140.